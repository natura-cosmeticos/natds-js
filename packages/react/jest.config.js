const base = require('../../jest.config.base.js')
const { displayName, name } = require('./package.json')

module.exports = {
  ...base,
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx}',
    '!src/**/*.stories.tsx',
    '!src/**/*.subcomponent.tsx',
    '!src/**/*.styles.ts',
    '!src/Templates/**',
    '!src/growthPlanColors/**',
    '!src/helpers/**'
  ],
  errorOnDeprecated: true,
  displayName,
  name,
  testEnvironment: 'jsdom',
  setupFilesAfterEnv: ['<rootDir>/setupTests.js'],
  roots: ['<rootDir>/src'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx',
    '\\.(jpg|jpeg|png|gif|webp|svg)$': 'jest-transform-file'
  }
}
