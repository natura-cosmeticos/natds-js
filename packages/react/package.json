{
  "name": "@naturacosmeticos/natds-react",
  "displayName": "NatDS-React",
  "description": "A collection of components from Natura Design System for React",
  "version": "2.66.0-alpha.DSY-5577.5.0",
  "private": false,
  "keywords": [
    "design-system",
    "natura",
    "react",
    "web"
  ],
  "homepage": "https://github.com/natura-cosmeticos/natds-js/tree/master/packages/react/README.md",
  "bugs": {
    "url": "https://github.com/natura-cosmeticos/natds-js/issues"
  },
  "license": "ISC",
  "author": {
    "email": "designsystem@natura.net",
    "name": "Natura&Co.",
    "url": "https://github.com/natura-cosmeticos"
  },
  "files": [
    "README.md",
    "dist/"
  ],
  "jsdelivr": "dist/umd/index.js",
  "source": "src/index.ts",
  "types": "dist/index.d.ts",
  "typings": "dist/index.d.ts",
  "umd:main": "dist/umd/index.js",
  "unpkg": "dist/umd/index.js",
  "main": "dist/index.js",
  "directories": {
    "lib": "dist"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/natura-cosmeticos/natds-js.git"
  },
  "scripts": {
    "build": "yarn clean && tsc -p ./tsconfig.json && yarn rollup:build",
    "build:storybook": "yarn build-storybook -o docs/dist",
    "build:storybook:ci": "yarn build:storybook --quiet",
    "clean": "rm -rf ./dist",
    "copy:assets": "mkdir -p ./dist/assets && cp -r ./src/assets/* ./dist/assets",
    "lint": "eslint --ext .js,.jsx,.ts,.tsx ./src --color",
    "lint:fix": "yarn lint --fix",
    "rollup:build": "rollup -c",
    "storybook": "start-storybook --no-manager-cache -p 6006",
    "test": "jest --testPathPattern=src --verbose",
    "test:ci": "jest  --testPathPattern=src --coverage && codecov -t",
    "test:regression": "loki",
    "test:watch": "jest --watch --detectOpenHandles",
    "test:clean": "jest --update-snapshot",
    "generate": "yarn plop --plopfile ./generators/plopfile.js"
  },
  "dependencies": {
    "@naturacosmeticos/natds-icons": "1.28.1",
    "@naturacosmeticos/natds-themes": "0.83.3",
    "react-jss": "10.9.0"
  },
  "devDependencies": {
    "@types/react": "17.0.39",
    "@types/react-dom": "17.0.13",
    "plop": "3.0.5",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-test-renderer": "17.0.2",
    "storybook-dark-mode": "1.0.9"
  },
  "peerDependencies": {
    "@types/react": ">= 16.8.6",
    "@types/react-dom": "17.0.13",
    "react": ">= 16.8.4",
    "react-dom": ">= 16.8.4"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    }
  }
}
