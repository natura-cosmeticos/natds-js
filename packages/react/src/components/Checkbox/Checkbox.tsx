/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { useRef } from 'react'
import { Ripple } from '../Ripple'
import { Label } from '../Label'
import { CheckboxProps } from './Checkbox.props'
import styles from './Checkbox.styles'

const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(({
  checked = false,
  className = '',
  disabled = false,
  id,
  brand,
  indeterminate = false,
  onChange,
  testID = `ds-checkbox-${id}`,
  value,
  label
}, ref) => {
  const {
    checkbox, wrapper, container, labelText
  } = styles({ indeterminate, disabled, brand })

  const getRippleColor = checked && !disabled ? 'primary' : 'highlight'
  const inputRef = useRef<HTMLInputElement>(null)

  const handleRippleClick = (event: React.MouseEvent<HTMLDivElement>) => {
    if (disabled) {
      return
    }

    if (onChange) {
      if (inputRef.current) {
        inputRef.current.focus()
      }
      onChange(event as any)
    }
  }

  return (
    <div className={`${className} ${container} Checkbox-gaya`}>
      <Ripple
        isCentered
        brand={brand}
        hideOverflow={false}
        disabled={disabled}
        color={getRippleColor}
        animationDuration={500}
        radius="50%"
        onClick={handleRippleClick}
        showHover
        focus
      >
        <div className={wrapper}>
          <input
            data-testid={testID}
            checked={checked}
            className={checkbox}
            disabled={disabled}
            id={id}
            onChange={onChange}
            type="checkbox"
            value={value}
            ref={ref || inputRef}
          />
        </div>
      </Ripple>
      {label && <Label className={labelText} label={label} htmlFor={id} />}
    </div>
  )
})

export default Checkbox
