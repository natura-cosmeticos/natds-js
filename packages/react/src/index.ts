export * from './ThemeProvider'
export * from './growthPlanColors/growthPlanColors'
export { useTheme, createUseStyles } from 'react-jss'

export * from './components/AppBarTop'
export * from './components/Avatar'
export * from './components/Badge'
export * from './components/Button'
export * from './components/Card'
export * from './components/Checkbox'
export * from './components/Collapse'
export * from './components/Counter'
export * from './components/Dialog'
export * from './components/Divider'
export * from './components/Drawer'
export * from './components/Icon'
export * from './components/IconButton'
export * from './components/Input'
export * from './components/Label'
export * from './components/Link'
export * from './components/ListItem'
export * from './components/Logo'
export * from './components/MenuItem'
export * from './components/ProgressIndicator'
export * from './components/RadioButton'
export * from './components/Rating'
export * from './components/Shortcut'
export * from './components/TextField'
export * from './components/Tab'
export * from './components/TabItem'
export * from './components/Alert'
export * from './components/Select'
export * from './components/Switch'
export * from './components/Tag'
export * from './components/Chip'
export * from './components/Typography'
export * from './components/AccordionGroup'
export * from './components/AccordionItem'
export * from './components/Image'
export * from './components/Snackbar'
export * from './components/Autocomplete'
export * from './components/GayaButton'
