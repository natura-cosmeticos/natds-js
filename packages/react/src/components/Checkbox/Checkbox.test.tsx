import React from 'react'
import { fireEvent } from '@testing-library/react'
import renderWithTheme from '../../helpers/renderWithTheme'
import Checkbox from './Checkbox'
import { CheckboxProps } from './Checkbox.props'

const defaultProps: CheckboxProps = {
  onChange: () => '',
  id: 'any',
  value: 'any value'
}

describe('Checkbox component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when is checked', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} checked />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when is indeterminate', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} indeterminate />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly when is disabled', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} disabled />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with label', () => {
    const { styles, component } = renderWithTheme(<Checkbox {...defaultProps} label="something" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should call onChange', () => {
    const onChangeMock = jest.fn()
    const { component } = renderWithTheme(<Checkbox {...defaultProps} onChange={onChangeMock} />)

    fireEvent.click(component.getByTestId('ds-checkbox-any'))

    expect(onChangeMock).toHaveBeenCalledTimes(2)
  })
})
