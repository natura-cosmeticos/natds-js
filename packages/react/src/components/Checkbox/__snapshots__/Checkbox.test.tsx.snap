// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render correctly when is checked 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: pointer;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    background-color: #f4ab34;
    border-radius: none;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #f4ab34;
    opacity: 0.16;
    border-radius: none;
  }
  .ripple-d1-0-2-14 {
    background-color: #f4ab34;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #8c8c8c;
    background-color: #8c8c8c;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .checkbox-d0-0-2-5 {  }
  .checkbox-d0-0-2-5:checked {
    background-color: #f4ab34;
    border: 2px solid #f4ab34;
  }
  .checkbox-d0-0-2-5:checked:after {
    content: \\"\\";
  }
  .labelText-d1-0-2-6 {
    color: #363636;
  }",
  <div>
    <div
      class=" container-0-2-1 Checkbox-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            checked=""
            class="checkbox-0-2-3 checkbox-d0-0-2-5"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly when is disabled 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: default;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    border-radius: none;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #111111;
    opacity: 0;
    border-radius: none;
  }
  .ripple-d1-0-2-14 {
    background-color: #111111;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #8c8c8c;
    background-color: #8c8c8c;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .checkbox-d0-0-2-5 {  }
  .checkbox-d0-0-2-5:checked {
    background-color: #f4ab34;
    border: 2px solid #f4ab34;
  }
  .checkbox-d0-0-2-5:checked:after {
    content: \\"\\";
  }
  .labelText-d1-0-2-6 {
    color: #8c8c8c;
  }",
  <div>
    <div
      class=" container-0-2-1 Checkbox-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-5"
            data-testid="ds-checkbox-any"
            disabled=""
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly when is indeterminate 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: pointer;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    background-color: #111111;
    border-radius: none;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #111111;
    opacity: 0.16;
    border-radius: none;
  }
  .ripple-d1-0-2-14 {
    background-color: #111111;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #8c8c8c;
    background-color: #8c8c8c;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .checkbox-d0-0-2-5 {  }
  .checkbox-d0-0-2-5:checked {
    background-color: #f4ab34;
    border: 2px solid #f4ab34;
  }
  .checkbox-d0-0-2-5:checked:after {
    content: \\"\\";
  }
  .labelText-d1-0-2-6 {
    color: #363636;
  }",
  <div>
    <div
      class=" container-0-2-1 Checkbox-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-5"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly with default props 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: pointer;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    background-color: #111111;
    border-radius: none;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #111111;
    opacity: 0.16;
    border-radius: none;
  }
  .ripple-d1-0-2-14 {
    background-color: #111111;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #8c8c8c;
    background-color: #8c8c8c;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .checkbox-d0-0-2-5 {  }
  .checkbox-d0-0-2-5:checked {
    background-color: #f4ab34;
    border: 2px solid #f4ab34;
  }
  .checkbox-d0-0-2-5:checked:after {
    content: \\"\\";
  }
  .labelText-d1-0-2-6 {
    color: #363636;
  }",
  <div>
    <div
      class=" container-0-2-1 Checkbox-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-5"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Checkbox component should render correctly with label 1`] = `
Array [
  ".sharedRippleEffect-0-2-7 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8 {
  height: fit-content;
  outline: none;
  position: relative;
}
.wrapper-0-2-8:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
}
.wrapper-0-2-8:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
}
.ripple-0-2-9 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-10 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-11 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-12 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-13 {
    cursor: pointer;
    display: inline-block;
    align-self: start;
  }
  .wrapper-d0-0-2-13:focus:after {
    background-color: #111111;
    border-radius: none;
  }
  .wrapper-d0-0-2-13:hover:after {
    background-color: #111111;
    opacity: 0.16;
    border-radius: none;
  }
  .ripple-d1-0-2-14 {
    background-color: #111111;
    height: 0;
    left: 50%;
    top: 50%;
    transition: scale opacity 500ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-15 {
    border-radius: 4px;
  }
  .rippleActive-d3-0-2-16 {
    animation: keyframes-ripple-0-2-12 500ms;
  }
  .labelContainer-0-2-17 {
    color: #111111;
    font-family: \\"Roboto\\", Arial;
  }
  .container-0-2-1 {
    display: flex;
    align-items: center;
  }
  .wrapper-0-2-2 {
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .checkbox-0-2-3 {
    width: 18px;
    border: 2px solid #737373;
    cursor: pointer;
    height: 18px;
    margin: 0;
    display: inline-block;
    position: relative;
    appearance: none;
    border-radius: 2px;
  }
  .checkbox-0-2-3:focus {
    outline: none;
  }
  .checkbox-0-2-3:disabled {
    border: 2px solid #8c8c8c;
    cursor: default;
  }
  .checkbox-0-2-3:checked {
    color: #ffffff;
  }
  .checkbox-0-2-3:checked:after {
    top: 50%;
    left: 50%;
    display: block;
    position: absolute;
    font-size: 18px;
    transform: translate(-50%, -50%);
    text-align: center;
    font-family: natds-icons;
  }
  .checkbox-0-2-3:checked:disabled {
    border: 2px solid #8c8c8c;
    background-color: #8c8c8c;
  }
  .labelText-0-2-4 {
    font-size: 14px;
    font-family: Roboto, Arial;
    font-weight: 400;
    line-height: 1.5;
    letter-spacing: 0.22px;
  }
  .checkbox-d0-0-2-5 {  }
  .checkbox-d0-0-2-5:checked {
    background-color: #f4ab34;
    border: 2px solid #f4ab34;
  }
  .checkbox-d0-0-2-5:checked:after {
    content: \\"\\";
  }
  .labelText-d1-0-2-6 {
    color: #363636;
  }",
  <div>
    <div
      class=" container-0-2-1 Checkbox-gaya"
    >
      <div
        class="wrapper-0-2-8 wrapper-d0-0-2-13"
        data-testid="ripple-wrapper"
        tabindex="0"
      >
        <div
          class="rippleContainer-0-2-10 rippleContainer-d2-0-2-15"
        >
          <div
            class="ripple-0-2-9 ripple-d1-0-2-14 "
            data-testid="ripple-animation"
          />
        </div>
        <div
          class="wrapper-0-2-2"
        >
          <input
            class="checkbox-0-2-3 checkbox-d0-0-2-5"
            data-testid="ds-checkbox-any"
            id="any"
            type="checkbox"
            value="any value"
          />
        </div>
      </div>
      <label
        class="labelText-0-2-4 labelText-d1-0-2-6 labelContainer-0-2-17"
        for="any"
      >
        something
      </label>
    </div>
  </div>,
]
`;
